src/Gui/View3DInventorViewer.cpp

```diff
@@ -2392,13 +2391,23 @@ void View3DInventorViewer::printDimension()
         short dimX, dimY;
         size.getValue(dimX, dimY);
 
-        float fHeight = static_cast<SoOrthographicCamera*>(getSoRenderManager()->getCamera())->height.getValue();
-        float fWidth = fHeight;
+        fHeight = static_cast<SoOrthographicCamera*>(getSoRenderManager()->getCamera())->height.getValue();
+        fWidth = fHeight;
 
         if (dimX > dimY)
-            fWidth *= ((float)dimX)/((float)dimY);
+            fWidth *= ((float)dimX) / ((float)dimY);
         else if (dimX < dimY)
-            fHeight *= ((float)dimY)/((float)dimX);
+            fHeight *= ((float)dimY) / ((float)dimX);
+    }
+}
+
+void View3DInventorViewer::printDimension()
+{
+    float fHeight = 0.0;
+    float fWidth = 0.0;
+    getDimensions(fHeight, fWidth);
+
+    if (!(fHeight == 0.0 || fWidth == 0.0)) {
```

Comment:
Why shouldn't the size be printed of width or height is 0.0?

Reply:
why not use `(fHeight != 0.0 && fWidth != 0.0)`

Reply:
What I wanted to do : 
```
float fHeight , fWidth;
    getDimensions(fHeight, fWidth);

    if (fHeight or fWidth initialized) {
```
So that if getDimensions doesn't work (if `t.isDerivedFrom` is false) then the printDimension doesn't print.

However now I can't see any problem removing this if completely. if `t.isDerivedFrom` is false it will just print 0.0 x 0.0.

Reply:
You can initialize fHeight and fWidth with -1.0 and then check if both are >= 0. Printing the dimensions of (0 mm, 0mm) in perspective mode should be avoided.

Reply:
While looking at View3DInventorViewer::printDimension() I have implemented it to also determine the size of a perspective camera: 45dc6d3a49

Reply:
Ok. So I'll have to rebase and solve conflict.